RATEST-223: Add functionality for editing Vitals

Irenyak1 opened the pull request:
Ticket - [RATEST-223](https://issues.openmrs.org/browse/RATEST-223)

Fixed  > > Add page functionality for Editing Patient Vitals and Editing Patient Contact Info

Irenyak1 commented:
> @Irenyak1 are you intentionally duplicating this page? https://github.com/openmrs/openmrs-distro-referenceapplication/blob/master/ui-tests/src/test/java/org/openmrs/reference/page/PatientCaptureVitalsPage.java

@kdaud , Can we have a call about this issue? I have tried making a work around to use the same page for editing vitals and it has failed.

sherrif10 commented on ui-tests/src/test/java/org/openmrs/reference/page/EditVitalsPage.java:
```diff
@@ -0,0 +1,134 @@
+
+public class EditVitalsPage extends Page {
+	
+	public static final String URL_PATH = "htmlformentryui/htmlform/editHtmlFormWithStandardUi.page";
```
Collect identation here . thanks this looks morethan good

sherrif10 commented on ui-tests/src/test/java/org/openmrs/reference/page/EditVitalsPage.java:
```diff
@@ -0,0 +1,134 @@
+    private static final By BLOOD_PRESSURE_FIELD_1 = By.id("w18");
+    private static final By BLOOD_PRESSURE_FIELD_2 = By.id("w20");
+    private static final By BLOOD_OXYGEN_SATURATION_FIELD = By.id("w22");
+	private static final By SAVE_CHANGES_BUTTON = By.xpath("//input[@type='button']");
```
Collect the indentation here

sherrif10 commented on ui-tests/src/test/java/org/openmrs/reference/page/EditVitalsPage.java:
```diff
@@ -0,0 +1,134 @@
+            respiratoryField.clear();
+            respiratoryField.sendKeys(value);
+            findElement(By.xpath("//ul[@id='formBreadcrumb']/li/ul/li[7]")).click();
+        } catch (Exception e) {}
```
Remove this catch blocks if they are not returning anything like it had been fixed [here ](https://github.com/openmrs/openmrs-distro-referenceapplication/pull/569/files). Ensure you are not breaking anything.

sherrif10 commented on ui-tests/src/test/java/org/openmrs/reference/page/EditVitalsPage.java:
```diff
@@ -0,0 +1,134 @@
+            temperatureField.clear();
+            temperatureField.sendKeys(value);
+            findElement(By.xpath("//ul[@id='formBreadcrumb']/li/ul/li[5]")).click();
+        } catch (Exception e) {}
```
Same here, Fee free to remove all since they are not necessary

kdaud commented:
EditVitals  is a page element on [PatientVisitsDashboardPage](https://github.com/openmrs/openmrs-distro-referenceapplication/blob/master/ui-tests/src/test/java/org/openmrs/reference/page/PatientVisitsDashboardPage.java) which is currently missing in distro. When the user clicks in the link, it opens edit Vitals form which uses the functionality of [PatientCaptureVitalsPage](https://github.com/openmrs/openmrs-distro-referenceapplication/blob/master/ui-tests/src/test/java/org/openmrs/reference/page/PatientCaptureVitalsPage.java). So we do not need to duplicate PatientCaptureVitalsPage but rather only adding the missing page element on PatientVisitsDashboard page.

Irenyak1 commented:
> EditVitals is a page-element on [PatientVisitsDashboardPage](https://github.com/openmrs/openmrs-distro-referenceapplication/blob/master/ui-tests/src/test/java/org/openmrs/reference/page/PatientVisitsDashboardPage.java) which is currently missing in distro. When the user clicks in the link, it opens edit Vitals form which uses the functionality of [PatientCaptureVitalsPage](https://github.com/openmrs/openmrs-distro-referenceapplication/blob/master/ui-tests/src/test/java/org/openmrs/reference/page/PatientCaptureVitalsPage.java). So we do not need to duplicate PatientCaptureVitalsPage but rather only adding the missing page element on PatientVisitsDashboard page.
> 
> ![Screenshot from 2021-10-08 09-24-30](https://user-images.githubusercontent.com/38831673/136509299-0bebf3d9-15db-4787-8708-851fcd14e117.png)

@kdaud I have actually realized that much as the PatientCaptureVitalsPage has functionality for editing vitals, it does not fully implement that functionality. I needed to create a separate Page for the test to successfully pass.

sherrif10 commented on ui-tests/src/test/java/org/openmrs/reference/page/EditVitalsPage.java:
```diff
@@ -0,0 +1,109 @@
+    }
+    
+    public void setBloodOxygenSaturationField(String value) {
+        WebElement bloodOxygenSaturationField = findElement(BLOOD_OXYGEN_SATURATION_FIELD);
```
Put this on the same line , otherwise thanks much

kdaud commented on ui-tests/src/test/java/org/openmrs/reference/page/ClinicianFacingPatientDashboardPage.java:
```diff
@@ -200,6 +204,21 @@ public PatientCaptureVitalsPage goToPatientCaptureVitalsPage() {
+    }
+    
+    public void deleteVitals() {
+        clickOn(DELETE_VITALS_ICON);
```
I have never know why Git sometimes makes the indention appear like that, but when the indention is pretty ok. Have just checked it behind the [seen](https://github.com/openmrs/openmrs-distro-referenceapplication/blob/a1d6a0d6218381952b1d0443f16b1e3b56e5ff0a/ui-tests/src/test/java/org/openmrs/reference/page/ClinicianFacingPatientDashboardPage.java#L213-L221) and observed what git is reporting at line `213 `to `215` sn't true.

kdaud commented:
@Irenyak1 are all [these](https://github.com/openmrs/openmrs-distro-referenceapplication/pull/584/files) changes?

sherrif10 commented:
Squash all those commits to a relatively maximum 6 commits :face_with_head_bandage:

Irenyak1 commented:
> Squash all those commits to a relatively maximum 6 commits face_with_head_bandage

@sherrif10 I have actually been squashing some commits. Only that I have moved back and forth as I was attending to the reviews. 
some commits are not squashed.

sherrif10 commented:
> some commits are not squashed.

Worry not about squashing, Prayers answered :face_with_head_bandage:

Irenyak1 commented:
> > you can always run git diff to be able see your changes before committing them.
> 
> @Irenyak1 have you had a look at your [changes](https://github.com/openmrs/openmrs-distro-referenceapplication/pull/584/files) before committing them?

Yeah I even did a git reset --hard master on my branch then I introduced my changes after which I added, committed and pushed. I actually I force pushed that branch hoping that it will overwrite these old changes. 
Is there anything different I could do to solve this? @kdaud

kdaud commented:
> @kdaud and it is the one I am using (JEE ) version.

@Irenyak1 doing the needful would be sweet!

Irenyak1 commented:
> > @kdaud and it is the one I am using (JEE ) version.
> 
> @Irenyak1 doing the needful would save time.

I think I am going to clone a fresh copy of the distro and apply my changes afresh, because the formatting is giving me a day mare.  I will have to close this PR.

Irenyak1 commented:
> > > @kdaud and it is the one I am using (JEE ) version.
> > 
> > 
> > @Irenyak1 doing the needful would save time.
> 
> I think I am going to clone a fresh copy of the distro and apply my changes afresh, because the formatting is giving me a day mare. I will have to close this PR.

Hello @kdaud , @sherrif10 , @dkayiwa  I have deleted the old copy of my distro and forked and cloned a fresh one, but I still have an issue. The Clinician facing patient Dashboard Page gets re written when I run the commit command yet there are only a few changes that I introduce to that page. Please this has blocked me for more that 3 hours. Can someone help me out?

Irenyak1 commented:
Hello @here I have migrated this PR to [here](https://github.com/openmrs/openmrs-distro-referenceapplication/pull/594) .
I was unblocked, thanks to @kdaud for the timely help and guidance. 
I can now proceed to another ticket. :+1:
